src/System.Security.Cryptography.Cng/src/Internal/Cryptography/Helpers.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+using ErrorCode = Interop.NCrypt.ErrorCode;
+
+namespace Internal.Cryptography
+{
+    internal static class Helpers
+    {
+        public static byte[] CloneByteArray(this byte[] src)
+        {
+            return (byte[])(src.Clone());
+        }
+
+        //
+        // The C# construct
+        //   
+        //    fixed (byte* p = new byte[0])
+        //
+        // sets "p" to 0 rather than a valid address. Sometimes, we actually want a non-NULL pointer instead. (Some CNG apis actually care whether the buffer pointer is
+        // NULL or not, even if the accompanying size argument is 0.)
+        //
+        // This helper enables the syntax:
+        //
+        //    fixed (byte* p = new byte[0].MapZeroLengthArrayToNonNullPointer())
+        //
+        // which always sets "p" to a non-NULL pointer for a non-null byte array. 
+        //
+        public static byte[] MapZeroLengthArrayToNonNullPointer(this byte[] src)
+        {
+            if (src != null && src.Length == 0)
+                return new byte[1];
```

Comment:
How often is this helper used?  If it's common, could we cache such an array in a read-only static?